### PR TITLE
Fix #10365: ImageCropper originalFilename for streamed content

### LINKED ISSUE #10365: ImageCropper: originalFilename does not match the name attribute of DefaultStreamedContent
### Describe the bug

`CroppedImage `object has an `originalFilename` attribute which is the name of the image that has been cropped.

If one is cropping an image which is obtained via a `StreamContent` rather than a static file, then there is no filename as such. Conveniently, a name can still be set to the `StreamContent`:

`StreamedContent imageAsStream = DefaultStreamedContent.builder()**.name("example")**.contentType("image/png").stream(() -> stream).build();`

yet, when one calls the method `getOriginalFileName `on the `CroppedImage`, it produces a `String `which is the result of the `toString()` method of the original `StreamContent`, instead of returning the `name` attribute that had been set in the builder.

Use case: I need to match the original pic with the cropped pic, and it would make my life easier if I could just use their names to match them.

### Reproducer

This case is deeply embedded in an app, can't create a reproducer.

### Expected behavior

The `originalFilename `attribute of `CroppedImage `should match the `name `attribute of the image that was cropped and which originated from a `DefaultStreamContent`.

### PrimeFaces edition

Community

### PrimeFaces version

12.0

### Theme

default theme

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

17

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #10365: ImageCropper originalFilename for streamed content

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java [hunk 1]
```diff
@@ -206,7 +206,20 @@ public Object getConvertedValue(FacesContext context, UIComponent component, Obj
...
+            }
+            else if (imageObject instanceof StreamedContent) {
+                StreamedContent streamedContentTmp = (StreamedContent) imageObject;
+                originalFileName = streamedContentTmp.getName();
+            }
+            else {
+                throw new IllegalArgumentException("ImageCropper 'image' must be either a String relative path or a StreamedObject.");
+            }
+
+            return new CroppedImage(originalFileName, croppedOutImage.toByteArray(), x, y, w, h);
```
Comment (Rapster) [inline primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:222]: Can it be done in createStreamedContentForCropping method?

**Comment by Rapster** [comment]
This can go in 13.0.1 no?

**Comment by melloware** [comment]
Definitely

**Comment by tandraschko** [comment]
+1

File: primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java (hunk 1 above)
Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:222]: @melloware doesn't seem you saw this coment, but why not simply call:
```java
 return new CroppedImage(stream.getName(), croppedOutImage.toByteArray(), x, y, w, h);`
```

**Comment by Rapster** [comment]
Whether it's a String or a StreamContent, it gets wrapped into a StreamContent, that's why I mentionned createStreamedContentForCropping

**Pull request merged**